lost/src/main/java/com/mapzen/android/lost/internal/FusedLocationProviderApiImpl.java

```diff
@@ -173,9 +178,24 @@ private void requestLocationUpdatesInternal(LocationRequest request) {
     }
   }
 
+  private void removeLocationUpdatesInternal(Set<LocationRequest> requests) {
+    if (requests == null) {
+      return;
+    }
+    for (LocationRequest request : requests) {
+      try {
+        service.removeLocationUpdates(request);
```

Comment:
Sending a separate request to the service for each request to be removed will cause a disable/enable cycle for each one. This has the side effect of sending out multiple location updates in rapid succession via `FusionEngine#checkLastKnownAndNotify(provider)`.

I think active location listeners would be protected from this effect if they have a `fastestInterval` associate with their request but it might be worth considering a batch removal operation.